core/rawdb/freezer_memory.go

```diff
@@ -389,6 +391,30 @@ func (f *MemoryFreezer) TruncateTail(tail uint64) (uint64, error) {
 	return old, nil
 }
 
+// TruncateTailBlocks discards all data below the provided block.
+// Only truncates blocks and receipts.
+func (f *MemoryFreezer) TruncateTailBlocks(tail uint64) (uint64, error) {
```

Comment:
Please integrate the prunable notion into TruncateTail and then remove this function